src/main/java/co/rsk/bitcoinj/script/RedeemScriptValidator.java

```diff
@@ -34,32 +35,40 @@ protected static boolean hasStandardRedeemScriptStructure(List<ScriptChunk> chun
                 return false;
             }
 
-            // First chunk must be an OP_N
-            if (!isOpN(chunks.get(0))) {
+            // First chunk must be a number for the threshold
+            ScriptChunk firstChunk = chunks.get(0);
+            if (!isN(firstChunk)) {
                 return false;
             }
 
-            // Second to last chunk must be an OP_N opcode too, and there should be
-            // that many data chunks (keys).
-            ScriptChunk secondToLastChunk = chunks.get(chunks.size() - 2);
-            if (!isOpN(secondToLastChunk)) {
+            int chunksSize = chunks.size();
+            ScriptChunk lastChunk = chunks.get(chunksSize - 1);
+            if (!isOpCheckMultiSig(lastChunk)) {
                 return false;
             }
 
-            int numKeys = Script.decodeFromOpN(secondToLastChunk.opcode);
-            if (numKeys < 1 || chunks.size() != numKeys + 3) { // numKeys + M + N + OP_CHECKMULTISIG
+            // Second to last chunk must be a number too,
+            // and there should be that many data chunks (keys).
+            int secondToLastChunkIndex = chunksSize - 2;
+            ScriptChunk secondToLastChunk = chunks.get(secondToLastChunkIndex);
+            if (!isN(secondToLastChunk)) {
                 return false;
             }
 
-            for (int i = 1; i < chunks.size() - 2; i++) {
+            int numKeys = decodeN(secondToLastChunk);
+            if (numKeys < 1 || chunksSize != numKeys + 3) { // numKeys + M + N + OP_CHECKMULTISIG
+                return false;
+            }
```

Comment:
test this case

Reply:
The `numKeys < 1` verification is unreachable as far as I could check.

At first, it decodes the second last chunk:
  
```
  public int decodeN() {
        if (isOpCode()) {
            return decodeOpN();
        }

        if (isN()) {
            return data[0];
        }

        throw new IllegalArgumentException("Cannot decode number from chunk");
    }
```
 
If the chunk is an OpCode it must be between OP_1 and OP_16; otherwise, the method `isN()` previously executed would give false by going through the `isOpcodeSmallNumber` execution branch.

If it is a pushdata content (a.k.a. isN), the method `isN()` previously executed contains the verification `data[0] >= 1`.

So, I am adding a test for the second part of the `numKeys < 1 || chunksSize != numKeys + 3`.